Civi/Core/CiviEventDispatcher.php

```diff
@@ -15,7 +17,9 @@
  *
  * @see \CRM_Utils_Hook
  */
-class CiviEventDispatcher extends EventDispatcher {
+class CiviEventDispatcher implements EventDispatcherInterface {
```

Comment:
Is there a reason we need an interface? And if so what are the reasons for using the symfony event dispatcher interface? Doesn't introduce the version problems we have?

Reply:
This looks good, I think it's fine to implement the EventDispatcherInterface as it allows us to keep the same contract for now without rewriting the functionality completely. We don't need to rewrite a custom interface to be 4.x compatible - no sense in writing our own interface as we don't know that we'll have to. It's entirely possible Symfony keep the same "contract" for this through 6.x

Reply:
Indeed as long as Symfony keeps the same contract across the different versions (and we can only prove the current situation).

Reply:
We may have to revisit this in the future if/when Symfony change things and it affects us. But the likelihood is that we'll be stuck on 4.x for a while and the 4.x -> 5.x upgrade path is simpler with fewer major deprecations.

Reply:
Some Unit tests failed without doing this implement as they were expecting something that was a EventDispatcherInterface  object